cmd/sidecar/main.go

```diff
@@ -140,47 +142,92 @@ func deleteBackupHandler(w http.ResponseWriter, req *http.Request) {
-	if err != nil {
-		log.Error(err, "failed to open log file")
+	if err := deleteBackup(req.Context(), &backupConf, backupName); err != nil {
+		log.Error(err, "failed to delete backup")
 		http.Error(w, "backup failed", http.StatusInternalServerError)
 		return
 	}
-	defer backupLog.Close()
-	logWriter := io.MultiWriter(backupLog, os.Stderr)
 
-	xbcloud := exec.CommandContext(req.Context(), "xbcloud", xb.XBCloudArgs(xb.XBCloudActionDelete, &backupConf)...)
+	log.Info("Backup deleted successfully", "destination", backupConf.Destination, "storage", backupConf.Type)
+}
+
+func deleteBackup(ctx context.Context, cfg *xtrabackup.BackupConfig, backupName string) error {
+	logWriter := io.Writer(os.Stderr)
+	if backupName != "" {
+		backupLog, err := os.OpenFile(filepath.Join(mysql.BackupLogDir, backupName+".log"), os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0666)
+		if err != nil {
+			return errors.Wrap(err, "failed to open log file")
+		}
+		defer backupLog.Close()
+		logWriter = io.MultiWriter(backupLog, os.Stderr)
+	}
+	xbcloud := exec.CommandContext(ctx, "xbcloud", xb.XBCloudArgs(xb.XBCloudActionDelete, cfg)...)
 	xbcloudErr, err := xbcloud.StderrPipe()
 	if err != nil {
-		log.Error(err, "xbcloud stderr pipe failed")
-		http.Error(w, "backup failed", http.StatusInternalServerError)
-		return
+		return errors.Wrap(err, "xbcloud stderr pipe failed")
 	}
 	defer xbcloudErr.Close()
 	log.Info(
 		"Deleting Backup",
-		"destination", backupConf.Destination,
-		"storage", backupConf.Type,
+		"destination", cfg.Destination,
+		"storage", cfg.Type,
 		"xbcloudCmd", sanitizeCmd(xbcloud),
 	)
 	if err := xbcloud.Start(); err != nil {
-		log.Error(err, "failed to start xbcloud")
-		http.Error(w, "backup failed", http.StatusInternalServerError)
-		return
+		return errors.Wrap(err, "failed to start xbcloud")
 	}
 
 	if _, err := io.Copy(logWriter, xbcloudErr); err != nil {
-		log.Error(err, "failed to copy xbcloud stderr")
-		http.Error(w, "backup failed", http.StatusInternalServerError)
-		return
+		return errors.Wrap(err, "failed to copy xbcloud stderr")
 	}
 
 	if err := xbcloud.Wait(); err != nil {
-		log.Error(err, "failed waiting for xbcloud to finish")
-		http.Error(w, "backup failed", http.StatusInternalServerError)
-		return
+		return errors.Wrap(err, "failed waiting for xbcloud to finish")
 	}
-	log.Info("Backup deleted successfully", "destination", backupConf.Destination, "storage", backupConf.Type)
+	return nil
+}
+
+func backupExists(ctx context.Context, cfg *xtrabackup.BackupConfig) (bool, error) {
+	storage, err := newStorage(cfg)
+	if err != nil {
+		return false, errors.Wrap(err, "new storage")
+	}
+	objects, err := storage.listObjects(ctx, cfg.Destination)
+	if err != nil {
+		return false, errors.Wrap(err, "list objects")
+	}
+	if len(objects) == 0 {
+		return false, nil
+	}
+	return true, nil
+}
+
+func checkBackupMD5Size(ctx context.Context, cfg *xtrabackup.BackupConfig) error {
+	// xbcloud doesn't create md5 file for azure
```

Comment:
@tplavcic Could you please check if PXB team has a task about it? Thanks.

Reply:
Now they have ;) https://jira.percona.com/browse/PXB-2952